File: Java/PrimerParcial/TP3/ej7/Account.java
package ar.itba.edu.POO.PrimerParcial.TP3.ej7;

public abstract class Account implements Comparable<Account>{
    public double balance;
    public int ID;

    public void deposit(double val){
        balance+=val;

    }
    protected abstract double extract(double val);

    public double getBalance(){
        return balance;
    }

    public int getID() {
        return ID;
    }

    @Override
    public String toString(){
        return String.format("Cuenta %d con saldo %.2f",ID,balance);
    }
}

File: Ruby/TP8/Ej5/account.rb
class Account
  def initialize(id)
    @id=id
  end
  def deposit(amount)
    @balance+=amount
  end

  def extract(amount)
    if can_extract(amount)
    @balance-=amount
    end
    @balance
    end

  def to_s
    'Cuenta '+@id.to_s+ ' con saldo ' + @balance.to_s
  end

  def get_balance
    @balance
  end

end
